subprojects/file-collections/src/main/java/org/gradle/api/internal/file/collections/MinimalFileTree.java

```diff
@@ -42,21 +39,6 @@
     void visitStructure(MinimalFileTreeStructureVisitor visitor, FileTreeInternal owner);
 
     interface MinimalFileTreeStructureVisitor {
-
-        /**
-         * Called prior to visiting a file source with the given spec, and allows this visitor to skip these files.
-         * A "file source" is some opaque source of files that is not a full {@link FileCollection}.
-         *
-         * <p>Note that this method is not necessarily called immediately before one of the visit methods, as some collections may be
-         * resolved in parallel. However, all visiting is performed sequentially and in order.
-         * This method is also called sequentially and in order.
-         *
-         * @return how should the collection be visited?
-         */
-        default FileCollectionStructureVisitor.VisitType prepareForVisit(FileCollectionInternal.Source source) {
-            return FileCollectionStructureVisitor.VisitType.Visit;
-        }
```

Comment:
Does this have anything to do with `ArtifactVisitor.prepareForVisit()`? Can we remove that as well maybe?

Reply:
`ArtifactVisitor.prepareForVisit()` is unrelated. This is only for `MinimalFileTree`, since none of the minimal file trees now need special handling when the visitor does not want to visit the contents.